src/codeview.jl

```diff
@@ -76,16 +76,22 @@ function is_type_unstable(code::Union{IRCode, CodeInfo}, idx::Int, used::BitSet)
     return (idx in used) && type isa Type && (!Base.isdispatchelem(type) || type == Core.Box)
 end
 
-cthulhu_warntype(args...) = cthulhu_warntype(stdout::IO, args...)
-function cthulhu_warntype(io::IO, src, rettype, debuginfo, stable_code, inline_cost=false)
+cthulhu_warntype(args...; kwargs...) = cthulhu_warntype(stdout::IO, args...; kwargs...)
+function cthulhu_warntype(io::IO, debuginfo::Union{DebugInfo,Symbol},
+    src::Union{CodeInfo,IRCode}, @nospecialize(rt), mi::Union{Nothing,MethodInstance} = nothing;
+    verbose::Bool=true, inline_cost::Bool=false)
     if inline_cost
-        error("Need a MethodInstance to show inlining costs. Call `cthulhu_typed` directly instead.")
+        @assert isa(mi, MethodInstance) "`mi::MethodInstance` is required"
```

Comment:
This probably shouldn't use `@assert` since IIUC, it is also meant to be called directly and there have been discussions about disabling `@assert` statements by default in the future.

Reply:
Thanks ! Now I just remember I got the same pointer from you on Julia Base 😅 Will make sure to use explicit error.